.github/workflows/build_and_functional_tests.yml

```diff
@@ -0,0 +1,78 @@
+name: Build and run functional tests using ragger through reusable workflow
+
+# This workflow will Lint, Build and then Run tests.
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - develop
+    tags:
+      - '*'
+  pull_request:
+
+jobs:
+  Build:
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Install dependencies
+        run: sudo apt update && sudo apt install -qy eslint
+        if: runner.os == 'Linux'
+
+      - name: Clone
+        uses: actions/checkout@v4
+
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: node
+
+      - name: Packages installation
+        run: npm install
+
+      - name: Compile
+        run: npm run compile
+
+      - name: Linter
+        if: runner.os == 'Linux'
+        run: npm run lint
+
+      - name: Test with Linux
+        if: runner.os == 'Linux'
+        run: xvfb-run -a npm test
+
+      - name: Test
+        if: runner.os != 'Linux'
+        run: npm test
+
+  Build_Windows:
+    strategy:
+      matrix:
+        os: [windows-latest]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Clone
+        uses: actions/checkout@v4
+
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: node
+
+      - name: Packages installation
+        run: cmd.exe /c npm.cmd install
+
+      - name: Compile
+        run: cmd.exe /c npm.cmd run compile
+
+      - name: Linter
+        run: cmd.exe /c npm.cmd run lint

```

Comment:
Since we run the linter on Linux, do we need to run it on windows as well ?

Reply:
It can get confusing if the linter on windows does not give the same result as the one on Linux...

Reply:
Yes, but in such case, it would mean something is not clear...

Reply:
Then we need to run the linter on macOS as well which is not the case : 

```
name: Linter
        if: runner.os == 'Linux'
        run: npm run lint
```

Reply:
oh, I misread my workflow. I thought it was already the case.
Then, Ok, to simplify, let's only run the linter on `ubuntu`.